enhancements/bare-metal-fulfillment/README.md

```diff
@@ -0,0 +1,267 @@
+      name: example
+    spec:
+      hostRequests:
+      - resourceClass: fc430
+        numberOfHosts: 2
+
+      # The networkAttachments section controls how networks are connected to
+      # physical interfaces.
+      networkAttachments:
+      # This configuration will match any available interface.
+      - primary: network1
+	    vlans:
+	    - network2
+      # This configuration will only apply to interfaces that match one of the
+      # listed MAC addresses (the tenant may not know the MAC address of the hosts until
+      # after they have acquired the host).
+      - primary: network3
+        matches:
+          macaddrs:
+          - C0:FF:EE:5D:57:7C
+          - C0:FF:EE:96:AD:CA
+      # This configuration will only match interfaces with the `25gb` tag
+      - matches:
+   	    tag: 25gb
+        primary: storage-network
+
+In the simplest case, the requester does not specify explicit network connectivity:
+
+    apiVersion: cloudkit.openshift.io/v1alpha1
+    kind: HostPool
+    metadata:
+      name: example
+    spec:
+      hostRequests:
+      - resourceClass: fc430
+        numberOfHosts: 2
+      networks:
+      - network1
+      - storage-network
+
+With no explicit connectivity, all of the specified networks will be connected to available interfaces as access ports. It is an error
+if there are more networks than available interfaces.
+
+#### Hosts
+
+Tenants do not create Hosts directly; instead, they are created as part of the HostPool fulfillment workflow. Once created,
+tenants can perform host-specific operations:
+
+* Power control
+* Serial console access
+* Inventory information
+
+To support these operations, we propose a new Hosts API. For example, we can implement power control as an attribute on the Host object:
+
+    apiVersion: cloudkit.openshift.io/v1alpha1
+    kind: Host
+    metadata:
+      name: example
+    spec:
+      powerState: PowerOn
+      serialConsole: Enabled
+
+### Implementation Details/Notes/Constraints
+
+#### ESI Playbooks
+
+The ESI API supports all of the needed bare metal functionality (the cluster fulfillment workflow already relies upon ESI
+playbooks for bare metal network configuration) . However, it is important to recognize that O-SAC will not use the ESI concept
+of tenancy; from an ESI perspective, hosts will always remain leased to the O-SAC project. As a result, we will need to introduce
+a new method of keeping track of O-SAC tenancy.
+
+One solution would be to rely upon Ironic’s extra field, which allows an ESI project to set arbitrary key/value pairs upon a
+specified host. Doing so would not be difficult; it simply creates additional considerations when implementing playbooks related
+to host assignment and unassignment.
+
+#### Serial Console Support
+
+There are multiple directions we could explore for serial console support:
+
+* ESI already supports serial console access through socat (using an Ironic feature exposed through an ESI proxy). We could simply
```

Comment:
Be good to understand tradeoffs; i.e., could we layer socat on top of other solutions; socat is a pretty generic mechanism.  Is there a consistency advantage to going a different route.

Reply:
Pulled out a short discussion

Reply:
I think we absolutely want to utilize something that looks a lot like existing streaming interfaces in kubernetes (either the pod exec functionality, or the kubevirt console access).

Reply:
I'm starting to feel like we may want to pull out console access into a separate enhancement request; I know Adrian's brought up questions relating to console access for VMs which suggest that maybe there's a unified solution we should be designing towards?